### ISSUE TITLE
Inserting an image resizes the canvas, even when the canvas is bigger than the image inserted

### ISSUE THREAD

**Issue by senaiboy** [viliusle/miniPaint#362]
Steps to recreate:

- File > New
- Change Width and Height to a large number
- File > Open > Open File
- Select a small image (eg a 50px by 50px icon)
- Canvas gets resized to the width and height of the image

It happens whether the inserted image is smaller or larger than the canvas. Of course, canvas being resized upwards isn't as big an issue as having the canvas shrunk to 50px by 50px.

However, when there is already something else on the canvas, inserting a new image will not resize the canvas.

**Comment by viliusle** [comment]
Yes, I know all these rules. It was added long time ago mainly for regular users, who opens 1 image and expects it to take all space, nothing else is left. 

I do not see it as a bug, but as strategy, which works well for most users, but can be annoying for small part of users.
I already received similar requests to fix it. 

What I currently planning is to let user disable this auto-resize feature in user settings. Options can be:
- Auto (always resize, by default)
- Disabled (never auto-resize)
- Smart (auto-resize, but if user defines canvas width and height manually - do not auto-resize)

p.s. this is just early ideas. Other opinions are welcome.

**Comment by senaiboy** [comment]
Honestly, I don't see the point of shrinking the canvas automatically, unless one _opens_ the image file and not insert an image onto a canvas.

Other software (offline and online, free and paid) do not do this, so I'd vote for option 2 as default.
